{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Bring in needed mods\n",
    "import numpy as np, pandas as pd, scipy.stats as ss\n",
    "import seaborn as sns, sys, os, glob\n",
    "from matplotlib import pyplot as plt\n",
    "sys.path.insert(0, '/home/croth/QTL-functions/SCRIPTS/')\n",
    "import snpy, qtlfunctions as qtlfun"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>B3502_A1</th>\n",
       "      <th>B3502_B1</th>\n",
       "      <th>B3502_B7</th>\n",
       "      <th>CF830</th>\n",
       "      <th>JEC21</th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Oldix</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Qual</th>\n",
       "      <th>...</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Type</th>\n",
       "      <th>Avedepth</th>\n",
       "      <th>Mindepth</th>\n",
       "      <th>Maxdepth</th>\n",
       "      <th>Averatio</th>\n",
       "      <th>Minratio</th>\n",
       "      <th>Maxratio</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>1</td>\n",
       "      <td>51</td>\n",
       "      <td>3681</td>\n",
       "      <td>3860.63</td>\n",
       "      <td>...</td>\n",
       "      <td>G.T</td>\n",
       "      <td>snp</td>\n",
       "      <td>270.6</td>\n",
       "      <td>183.0</td>\n",
       "      <td>389.0</td>\n",
       "      <td>0.572585</td>\n",
       "      <td>0.483696</td>\n",
       "      <td>0.665254</td>\n",
       "      <td>G</td>\n",
       "      <td>G.T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1062</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1062</td>\n",
       "      <td>55108</td>\n",
       "      <td>90277.00</td>\n",
       "      <td>...</td>\n",
       "      <td>CGTTGGTTTTT.GGTTTTTTTTT.GGTTGTTTTT.GGTTGTTTTTT...</td>\n",
       "      <td>complex</td>\n",
       "      <td>62.6</td>\n",
       "      <td>31.0</td>\n",
       "      <td>115.0</td>\n",
       "      <td>0.240623</td>\n",
       "      <td>0.051724</td>\n",
       "      <td>0.937500</td>\n",
       "      <td>CGTTGGTTTTT</td>\n",
       "      <td>CGTTGGTTTTT.GGTTTTTTTTT.GGTTGTTTTT.GGTTGTTTTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8887</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>1</td>\n",
       "      <td>8887</td>\n",
       "      <td>461149</td>\n",
       "      <td>4093.03</td>\n",
       "      <td>...</td>\n",
       "      <td>G.C</td>\n",
       "      <td>snp</td>\n",
       "      <td>131.8</td>\n",
       "      <td>70.0</td>\n",
       "      <td>253.0</td>\n",
       "      <td>0.704192</td>\n",
       "      <td>0.575000</td>\n",
       "      <td>0.874317</td>\n",
       "      <td>G</td>\n",
       "      <td>G.C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10062</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>1</td>\n",
       "      <td>10062</td>\n",
       "      <td>530127</td>\n",
       "      <td>103609.00</td>\n",
       "      <td>...</td>\n",
       "      <td>C.G</td>\n",
       "      <td>snp</td>\n",
       "      <td>225.6</td>\n",
       "      <td>180.0</td>\n",
       "      <td>251.0</td>\n",
       "      <td>0.995530</td>\n",
       "      <td>0.994475</td>\n",
       "      <td>0.996032</td>\n",
       "      <td>C</td>\n",
       "      <td>C.G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11653</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>1</td>\n",
       "      <td>11653</td>\n",
       "      <td>636710</td>\n",
       "      <td>88263.90</td>\n",
       "      <td>...</td>\n",
       "      <td>TGGGGGGGGGGGGGGGGGGGTTGTGCTGATG.GGGGGGGGGGGGGG...</td>\n",
       "      <td>snp</td>\n",
       "      <td>7.8</td>\n",
       "      <td>3.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>0.014286</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.071429</td>\n",
       "      <td>TGGGGGGGGGGGGGGGGGGGTTGTGCTGATG</td>\n",
       "      <td>TGGGGGGGGGGGGGGGGGGGTTGTGCTGATG.GGGGGGGGGGGGGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       B3502_A1  B3502_B1  B3502_B7  CF830  JEC21       Seqid  Chrom  Oldix  \\\n",
       "51          0.0       1.0       1.0    0.0    0.0  AE017341.1      1     51   \n",
       "1062        1.0       1.0       4.0    1.0    1.0  AE017341.1      1   1062   \n",
       "8887        0.0       1.0       0.0    0.0    0.0  AE017341.1      1   8887   \n",
       "10062       0.0       0.0       0.0    1.0    1.0  AE017341.1      1  10062   \n",
       "11653       1.0       1.0       1.0    4.0    1.0  AE017341.1      1  11653   \n",
       "\n",
       "          Pos       Qual  ...  \\\n",
       "51       3681    3860.63  ...   \n",
       "1062    55108   90277.00  ...   \n",
       "8887   461149    4093.03  ...   \n",
       "10062  530127  103609.00  ...   \n",
       "11653  636710   88263.90  ...   \n",
       "\n",
       "                                                 Alleles     Type Avedepth  \\\n",
       "51                                                   G.T      snp    270.6   \n",
       "1062   CGTTGGTTTTT.GGTTTTTTTTT.GGTTGTTTTT.GGTTGTTTTTT...  complex     62.6   \n",
       "8887                                                 G.C      snp    131.8   \n",
       "10062                                                C.G      snp    225.6   \n",
       "11653  TGGGGGGGGGGGGGGGGGGGTTGTGCTGATG.GGGGGGGGGGGGGG...      snp      7.8   \n",
       "\n",
       "       Mindepth  Maxdepth  Averatio  Minratio  Maxratio  \\\n",
       "51        183.0     389.0  0.572585  0.483696  0.665254   \n",
       "1062       31.0     115.0  0.240623  0.051724  0.937500   \n",
       "8887       70.0     253.0  0.704192  0.575000  0.874317   \n",
       "10062     180.0     251.0  0.995530  0.994475  0.996032   \n",
       "11653       3.0      13.0  0.014286  0.000000  0.071429   \n",
       "\n",
       "                                   Ref  \\\n",
       "51                                   G   \n",
       "1062                       CGTTGGTTTTT   \n",
       "8887                                 G   \n",
       "10062                                C   \n",
       "11653  TGGGGGGGGGGGGGGGGGGGTTGTGCTGATG   \n",
       "\n",
       "                                                     Alt  \n",
       "51                                                   G.T  \n",
       "1062   CGTTGGTTTTT.GGTTTTTTTTT.GGTTGTTTTT.GGTTGTTTTTT...  \n",
       "8887                                                 G.C  \n",
       "10062                                                C.G  \n",
       "11653  TGGGGGGGGGGGGGGGGGGGTTGTGCTGATG.GGGGGGGGGGGGGG...  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Set path to GFF file\n",
    "gffpath ='/home/croth/Downloads/B3502/REF/FungiDB-48_CneoformansJEC21.gff.gz'\n",
    "\n",
    "## Set path to reference file\n",
    "refpath = '/home/croth/Downloads/B3502/REF/FungiDB-48_CneoformansJEC21_Genome.fasta'\n",
    "\n",
    "## Set genotype data path\n",
    "geno_path = '../GENOTYPE/B3502_parent_variants.csv.gz'\n",
    "\n",
    "## Bring in genotype data\n",
    "genos = pd.read_csv(geno_path,index_col=0)\n",
    "genos['Ref'] = [p.split('.')[0] for  p in genos.Alleles]\n",
    "genos['Alt'] = genos.Alleles\n",
    "genos.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(['B3502_A1', 'B3502_B1', 'B3502_B7', 'CF830', 'JEC21'], 16)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Gather samples\n",
    "samples = genos.columns[:5].tolist()\n",
    "\n",
    "## Gather info columns\n",
    "info_cols = [c for c in genos.columns if c not in samples]\n",
    "\n",
    "## View samples and # of info columns\n",
    "samples,len(info_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Length</th>\n",
       "      <th>Cumlen</th>\n",
       "      <th>Midpts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2300473</td>\n",
       "      <td>0</td>\n",
       "      <td>1150236.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>1495312</td>\n",
       "      <td>2300473</td>\n",
       "      <td>3048129.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>2082458</td>\n",
       "      <td>3795785</td>\n",
       "      <td>4837014.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>1783042</td>\n",
       "      <td>5878243</td>\n",
       "      <td>6769764.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>1505348</td>\n",
       "      <td>7661285</td>\n",
       "      <td>8413959.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Chrom   Length   Cumlen     Midpts\n",
       "0      1  2300473        0  1150236.5\n",
       "1      2  1495312  2300473  3048129.0\n",
       "2      3  2082458  3795785  4837014.0\n",
       "3      4  1783042  5878243  6769764.0\n",
       "4      5  1505348  7661285  8413959.0"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Make a chromosome map\n",
    "chrommap = qtlfun.chrommap(genos)\n",
    "chrommap.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(468,\n",
       " array(['../GENES/B3502_A1/CNC07110-t26_1.csv.gz',\n",
       "        '../GENES/B3502_A1/CND05940-t26_1.csv.gz',\n",
       "        '../GENES/B3502_A1/CNF04140-t26_1.csv.gz',\n",
       "        '../GENES/B3502_A1/CNB04980-t26_1.csv.gz',\n",
       "        '../GENES/B3502_A1/CNB00100-t26_1.csv.gz'], dtype='<U39'))"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Gather paths to gene dataframes\n",
    "samplespath = np.concatenate([glob.glob('../GENES/%s/C*-t26*.csv.gz'%s) for s in samples])\n",
    "len(samplespath),samplespath[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(468, 14)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Calculate differences for transcipts\n",
    "res = snpy.snpresults(samplespath)\n",
    "res.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(53,\n",
       " array(['../GENES/B3502_A1/CNE03000.mRNA.csv.gz',\n",
       "        '../GENES/B3502_A1/CNM00530.mRNA.csv.gz',\n",
       "        '../GENES/B3502_A1/CNB04970.mRNA.csv.gz',\n",
       "        '../GENES/B3502_A1/CND00840.mRNA.csv.gz',\n",
       "        '../GENES/B3502_A1/CNA03640.mRNA.csv.gz'], dtype='<U38'))"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Gather paths to gene dataframes of mrna\n",
    "samplespath_mrna = np.concatenate([glob.glob('../GENES/%s/C*mRNA*.csv.gz'%s) for s in samples])\n",
    "len(samplespath_mrna),samplespath_mrna[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(53, 14)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Calculate differences for mRNA \n",
    "res_m = snpy.snpresults(samplespath_mrna,parent_end='.mRNA')\n",
    "res_m.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Expected</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Refstop</th>\n",
       "      <th>Altstop</th>\n",
       "      <th>Nonsyn</th>\n",
       "      <th>Nvars</th>\n",
       "      <th>Utr5</th>\n",
       "      <th>Utr3</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Intron</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNE03000</td>\n",
       "      <td>CNE03000.mRNA</td>\n",
       "      <td>2153.000000</td>\n",
       "      <td>453</td>\n",
       "      <td>453</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNM00530</td>\n",
       "      <td>CNM00530.mRNA</td>\n",
       "      <td>1577.000000</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>7</td>\n",
       "      <td>66</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNB04970</td>\n",
       "      <td>CNB04970.mRNA</td>\n",
       "      <td>1031.000000</td>\n",
       "      <td>90</td>\n",
       "      <td>155</td>\n",
       "      <td>44</td>\n",
       "      <td>36</td>\n",
       "      <td>2</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CND00840</td>\n",
       "      <td>CND00840.mRNA</td>\n",
       "      <td>497.666667</td>\n",
       "      <td>37</td>\n",
       "      <td>37</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNA03640</td>\n",
       "      <td>CNA03640.mRNA</td>\n",
       "      <td>2111.000000</td>\n",
       "      <td>63</td>\n",
       "      <td>63</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Strain      Gene         Parent     Expected  Ref  Alt  Refstop  Altstop  \\\n",
       "0  B3502_A1  CNE03000  CNE03000.mRNA  2153.000000  453  453        7        7   \n",
       "1  B3502_A1  CNM00530  CNM00530.mRNA  1577.000000   29   29        7       66   \n",
       "2  B3502_A1  CNB04970  CNB04970.mRNA  1031.000000   90  155       44       36   \n",
       "3  B3502_A1  CND00840  CND00840.mRNA   497.666667   37   37       15       15   \n",
       "4  B3502_A1  CNA03640  CNA03640.mRNA  2111.000000   63   63       69       69   \n",
       "\n",
       "   Nonsyn  Nvars  Utr5  Utr3  Exon  Intron  \n",
       "0       0      2     0     0     2       0  \n",
       "1       0      2     0     0     2       0  \n",
       "2       2     23     0     0    23       0  \n",
       "3       0      2     0     0     2       0  \n",
       "4       0      1     0     0     1       0  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res_m.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Expected</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Refstop</th>\n",
       "      <th>Altstop</th>\n",
       "      <th>Nonsyn</th>\n",
       "      <th>Nvars</th>\n",
       "      <th>Utr5</th>\n",
       "      <th>Utr3</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Intron</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNF04940</td>\n",
       "      <td>CNF04940-t26_1</td>\n",
       "      <td>526.0</td>\n",
       "      <td>526</td>\n",
       "      <td>526</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Strain      Gene          Parent  Expected  Ref  Alt  Refstop  Altstop  \\\n",
       "151  B3502_B1  CNF04940  CNF04940-t26_1     526.0  526  526        1        1   \n",
       "\n",
       "     Nonsyn  Nvars  Utr5  Utr3  Exon  Intron  \n",
       "151       1     14     0     1    13       0  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Candidate repressor\n",
    "res[(res.Gene=='CNF04940')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Expected</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Refstop</th>\n",
       "      <th>Altstop</th>\n",
       "      <th>Nonsyn</th>\n",
       "      <th>Nvars</th>\n",
       "      <th>Utr5</th>\n",
       "      <th>Utr3</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Intron</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNN01270</td>\n",
       "      <td>CNN01270-t26_1</td>\n",
       "      <td>691.0</td>\n",
       "      <td>691</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNN01270</td>\n",
       "      <td>CNN01270-t26_1</td>\n",
       "      <td>691.0</td>\n",
       "      <td>691</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>B3502_B7</td>\n",
       "      <td>CNN01270</td>\n",
       "      <td>CNN01270-t26_1</td>\n",
       "      <td>691.0</td>\n",
       "      <td>691</td>\n",
       "      <td>456</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Strain      Gene          Parent  Expected  Ref  Alt  Refstop  Altstop  \\\n",
       "119  B3502_A1  CNN01270  CNN01270-t26_1     691.0  691  456        1        2   \n",
       "247  B3502_B1  CNN01270  CNN01270-t26_1     691.0  691  456        1        2   \n",
       "371  B3502_B7  CNN01270  CNN01270-t26_1     691.0  691  456        1        2   \n",
       "\n",
       "     Nonsyn  Nvars  Utr5  Utr3  Exon  Intron  \n",
       "119       0      1     0     0     1       0  \n",
       "247       0      1     0     0     1       0  \n",
       "371       0      1     0     0     1       0  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## RIC8\n",
    "res[(res.Gene=='CNN01270')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Expected</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Refstop</th>\n",
       "      <th>Altstop</th>\n",
       "      <th>Nonsyn</th>\n",
       "      <th>Nvars</th>\n",
       "      <th>Utr5</th>\n",
       "      <th>Utr3</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Intron</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNC06490</td>\n",
       "      <td>CNC06490-t26_1</td>\n",
       "      <td>996.0</td>\n",
       "      <td>996</td>\n",
       "      <td>996</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNC06490</td>\n",
       "      <td>CNC06490-t26_1</td>\n",
       "      <td>996.0</td>\n",
       "      <td>996</td>\n",
       "      <td>996</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>B3502_B7</td>\n",
       "      <td>CNC06490</td>\n",
       "      <td>CNC06490-t26_1</td>\n",
       "      <td>996.0</td>\n",
       "      <td>996</td>\n",
       "      <td>996</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Strain      Gene          Parent  Expected  Ref  Alt  Refstop  Altstop  \\\n",
       "12   B3502_A1  CNC06490  CNC06490-t26_1     996.0  996  996        1        1   \n",
       "141  B3502_B1  CNC06490  CNC06490-t26_1     996.0  996  996        1        1   \n",
       "272  B3502_B7  CNC06490  CNC06490-t26_1     996.0  996  996        1        1   \n",
       "\n",
       "     Nonsyn  Nvars  Utr5  Utr3  Exon  Intron  \n",
       "12        1      1     0     0     1       0  \n",
       "141       1      1     0     0     1       0  \n",
       "272       1      1     0     0     1       0  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## RHO\n",
    "res[(res.Gene=='CNC06490')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "repressors = ['CNA03680', 'CNB00860', 'CNC07030', 'CNC07110', 'CNE05380',\n",
    "               'CNF04940', 'CNL04090', 'CNM00880', 'CNM01500', 'CNN02460']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Strain</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Parent</th>\n",
       "      <th>Expected</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Refstop</th>\n",
       "      <th>Altstop</th>\n",
       "      <th>Nonsyn</th>\n",
       "      <th>Nvars</th>\n",
       "      <th>Utr5</th>\n",
       "      <th>Utr3</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Intron</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNC07110</td>\n",
       "      <td>CNC07110-t26_1</td>\n",
       "      <td>256.0</td>\n",
       "      <td>256</td>\n",
       "      <td>256</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNB00860</td>\n",
       "      <td>CNB00860-t26_1</td>\n",
       "      <td>361.0</td>\n",
       "      <td>361</td>\n",
       "      <td>361</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNC07030</td>\n",
       "      <td>CNC07030-t26_1</td>\n",
       "      <td>1588.0</td>\n",
       "      <td>1588</td>\n",
       "      <td>1574</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>608</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>B3502_A1</td>\n",
       "      <td>CNM01500</td>\n",
       "      <td>CNM01500-t26_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1120</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNN02460</td>\n",
       "      <td>CNN02460-t26_1</td>\n",
       "      <td>621.0</td>\n",
       "      <td>621</td>\n",
       "      <td>621</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNF04940</td>\n",
       "      <td>CNF04940-t26_1</td>\n",
       "      <td>526.0</td>\n",
       "      <td>526</td>\n",
       "      <td>526</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNM00880</td>\n",
       "      <td>CNM00880-t26_1</td>\n",
       "      <td>343.0</td>\n",
       "      <td>343</td>\n",
       "      <td>343</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNL04090</td>\n",
       "      <td>CNL04090-t26_1</td>\n",
       "      <td>1477.0</td>\n",
       "      <td>1477</td>\n",
       "      <td>1477</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNC07030</td>\n",
       "      <td>CNC07030-t26_1</td>\n",
       "      <td>1588.0</td>\n",
       "      <td>1588</td>\n",
       "      <td>1587</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>586</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNM01500</td>\n",
       "      <td>CNM01500-t26_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1120</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>B3502_B1</td>\n",
       "      <td>CNE05380</td>\n",
       "      <td>CNE05380-t26_1</td>\n",
       "      <td>938.0</td>\n",
       "      <td>938</td>\n",
       "      <td>938</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>B3502_B7</td>\n",
       "      <td>CNC07110</td>\n",
       "      <td>CNC07110-t26_1</td>\n",
       "      <td>256.0</td>\n",
       "      <td>256</td>\n",
       "      <td>256</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>B3502_B7</td>\n",
       "      <td>CNB00860</td>\n",
       "      <td>CNB00860-t26_1</td>\n",
       "      <td>361.0</td>\n",
       "      <td>361</td>\n",
       "      <td>361</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>B3502_B7</td>\n",
       "      <td>CNM01500</td>\n",
       "      <td>CNM01500-t26_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1120</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393</th>\n",
       "      <td>CF830</td>\n",
       "      <td>CNB00860</td>\n",
       "      <td>CNB00860-t26_1</td>\n",
       "      <td>361.0</td>\n",
       "      <td>361</td>\n",
       "      <td>361</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>CF830</td>\n",
       "      <td>CNM01500</td>\n",
       "      <td>CNM01500-t26_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1120</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>JEC21</td>\n",
       "      <td>CNC07110</td>\n",
       "      <td>CNC07110-t26_1</td>\n",
       "      <td>256.0</td>\n",
       "      <td>256</td>\n",
       "      <td>256</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>JEC21</td>\n",
       "      <td>CNB00860</td>\n",
       "      <td>CNB00860-t26_1</td>\n",
       "      <td>361.0</td>\n",
       "      <td>361</td>\n",
       "      <td>361</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>JEC21</td>\n",
       "      <td>CNM01500</td>\n",
       "      <td>CNM01500-t26_1</td>\n",
       "      <td>1120.0</td>\n",
       "      <td>1120</td>\n",
       "      <td>1120</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Strain      Gene          Parent  Expected   Ref   Alt  Refstop  \\\n",
       "0    B3502_A1  CNC07110  CNC07110-t26_1     256.0   256   256        1   \n",
       "37   B3502_A1  CNB00860  CNB00860-t26_1     361.0   361   361        1   \n",
       "85   B3502_A1  CNC07030  CNC07030-t26_1    1588.0  1588  1574        1   \n",
       "112  B3502_A1  CNM01500  CNM01500-t26_1    1120.0  1120  1120        1   \n",
       "145  B3502_B1  CNN02460  CNN02460-t26_1     621.0   621   621        1   \n",
       "151  B3502_B1  CNF04940  CNF04940-t26_1     526.0   526   526        1   \n",
       "167  B3502_B1  CNM00880  CNM00880-t26_1     343.0   343   343        1   \n",
       "209  B3502_B1  CNL04090  CNL04090-t26_1    1477.0  1477  1477        1   \n",
       "210  B3502_B1  CNC07030  CNC07030-t26_1    1588.0  1588  1587        1   \n",
       "237  B3502_B1  CNM01500  CNM01500-t26_1    1120.0  1120  1120        1   \n",
       "243  B3502_B1  CNE05380  CNE05380-t26_1     938.0   938   938        1   \n",
       "257  B3502_B7  CNC07110  CNC07110-t26_1     256.0   256   256        1   \n",
       "296  B3502_B7  CNB00860  CNB00860-t26_1     361.0   361   361        1   \n",
       "364  B3502_B7  CNM01500  CNM01500-t26_1    1120.0  1120  1120        1   \n",
       "393     CF830  CNB00860  CNB00860-t26_1     361.0   361   361        1   \n",
       "421     CF830  CNM01500  CNM01500-t26_1    1120.0  1120  1120        1   \n",
       "427     JEC21  CNC07110  CNC07110-t26_1     256.0   256   256        1   \n",
       "440     JEC21  CNB00860  CNB00860-t26_1     361.0   361   361        1   \n",
       "463     JEC21  CNM01500  CNM01500-t26_1    1120.0  1120  1120        1   \n",
       "\n",
       "     Altstop  Nonsyn  Nvars  Utr5  Utr3  Exon  Intron  \n",
       "0          1       0      1     1     0     0       0  \n",
       "37         1       0      1     0     1     0       0  \n",
       "85         1     608     42     0     0    42       0  \n",
       "112        1       0      3     2     0     0       1  \n",
       "145        1       3      5     0     0     5       0  \n",
       "151        1       1     14     0     1    13       0  \n",
       "167        1       0      3     0     3     0       0  \n",
       "209        1       0      1     0     1     0       0  \n",
       "210        1     586      3     0     0     3       0  \n",
       "237        1       0      5     4     0     0       1  \n",
       "243        1       0      1     0     0     1       0  \n",
       "257        1       0      1     1     0     0       0  \n",
       "296        1       0      1     0     1     0       0  \n",
       "364        1       0      3     2     0     0       1  \n",
       "393        1       0      1     0     1     0       0  \n",
       "421        1       0      5     4     0     0       1  \n",
       "427        1       0      1     1     0     0       0  \n",
       "440        1       0      1     0     1     0       0  \n",
       "463        1       0      3     2     0     0       1  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res[(res.Gene.isin(repressors))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "ename": "AssertionError",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-e99f91a18d62>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32massert\u001b[0m \u001b[0;36m1\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mAssertionError\u001b[0m: "
     ]
    }
   ],
   "source": [
    "assert 1 == 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Bring in chrommap\n",
    "chrommap = pd.read_csv('/home/croth/Downloads/B3502/DATA/chrommap.csv.gz')\n",
    "chrommap['Seqid'] = chrommap['Contig']\n",
    "chrommap['Chromosome'] = chrommap.index+1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Bring in GFF file\n",
    "gffpath ='/home/croth/Downloads/B3502/REF/FungiDB-48_CneoformansJEC21.gff.gz'\n",
    "\n",
    "names = [\"Seqid\", \"Source\", \"Type\", \n",
    "         \"Start\", \"End\", \"Score\", \n",
    "         \"Strand\", \"Phase\", \"Attributes\"]\n",
    "\n",
    "descriptions = ['hypothetical protein','unspecified product']\n",
    "dtype = [\"str\",\"str\",\"str\",\"int\",\"int\",\"str\",\"str\",\"str\",\"str\"]\n",
    "\n",
    "gff = pd.read_csv(gffpath,\n",
    "                   comment=\"#\",header=None,\n",
    "                  delimiter=\"\\t\",names=names,\n",
    "                    dtype=dict(zip(names,dtype)))\n",
    "\n",
    "foi = ['gene','three_prime_UTR','five_prime_UTR','CDS']\n",
    "\n",
    "gff = gff[(gff.Type.isin(foi))]\n",
    "\n",
    "gff['Parent'] = [a.split('Parent=')[-1].split(';')[0].split('ID=')[-1] \n",
    "                 for a in gff.Attributes ]\n",
    "\n",
    "gff['Gene'] = [a.split(';')[0].split('D=')[-1].split('-')[0]\n",
    "               for a in gff.Attributes]\n",
    "\n",
    "genes = gff[(gff.Type=='gene')].copy()\n",
    "genes['Description'] = [a.split('description=')[-1].split('%2C')[0] \n",
    "                        for a in genes.Attributes]\n",
    "\n",
    "genes = genes.merge(chrommap[['Seqid','Length','Chromosome','Midpts','Cumlen']])\n",
    "genes.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "B3502 = sorted(glob.glob('../GENES/B3502*/*.csv.gz')) \n",
    "CF830 = sorted(glob.glob('../GENES/CF830/*.csv.gz'))\n",
    "JEC21 = sorted(glob.glob('../GENES/JEC21/*.csv.gz'))\n",
    "#JEC20 = sorted(glob.glob('../GENES/JEC20/*.csv.gz'))\n",
    "samplespath = B3502+CF830+JEC21\n",
    "samplespath[:5],len(samplespath)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "todf = []\n",
    "for s in samplespath:\n",
    "\n",
    "    sample = s.split('/')[2]\n",
    "    genep = s.split('/')[-1].split('.csv')[0]\n",
    "    gene_name = s.split('/')[-1].split('-t26')[0]\n",
    "\n",
    "    temp = pd.read_csv(s)\n",
    "    gene = temp.Gene.min()\n",
    "    assert sample == temp.Sample.min()\n",
    "    assert gene == genep\n",
    "        \n",
    "    cds = temp[(temp.Type==0)]\n",
    "        \n",
    "    ref,alt = makeorf(cds)\n",
    "        \n",
    "    el = cds.Pos.unique().shape[0]/3-1\n",
    "        \n",
    "    ra = ref.translate(to_stop=True)\n",
    "    aa = alt.translate(to_stop=True)\n",
    "        \n",
    "    rl = len(ra)\n",
    "    al = len(aa)\n",
    "        \n",
    "    sr = ref.translate().count('*')\n",
    "    sa = alt.translate().count('*')\n",
    "        \n",
    "    ns = sa - 1\n",
    "    for i in range(np.min([len(ra),len(aa)])):\n",
    "        if ra[i]!=aa[i]:\n",
    "            ns = ns + 1\n",
    "            \n",
    "    nvars = temp[(temp.Isvar==1)].shape[0]\n",
    "    utr3 = temp[(temp.Type==3) & (temp.Isvar==1)].shape[0]\n",
    "    utr5 = temp[(temp.Type==5) & (temp.Isvar==1)].shape[0]\n",
    "    inexon = temp[(temp.Type==0) & (temp.Isvar==1)].shape[0]\n",
    "    inintron = temp[(temp.Type==-1) & (temp.Isvar==1)].shape[0]\n",
    "        \n",
    "    todf.append((sample,gene_name,gene,\n",
    "                 el,rl,al,sr,sa,ns,\n",
    "                 nvars,utr5,utr3,inexon,inintron\n",
    "                ))\n",
    "    \n",
    "    resdf = pd.DataFrame(todf,\n",
    "        columns=['Strain','Gene','Parent','Expected',\n",
    "                 'Ref','Alt','Refstop','Altstop','Nonsyn',\n",
    "                 'Nvars','Utr5','Utr3','Exon','Intron'])\n",
    "    \n",
    "    return resdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "resdf.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf.Gene.unique().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Nonsyn>0)].Gene.unique().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Utr5>0)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Utr3>0)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Altstop==0)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Altstop>1)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Refstop==0)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Refstop>1)].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "toredo = resdf[(resdf.Expected!=resdf.Ref)]\n",
    "toredo.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf.Strain.unique().shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Genes with stop gains within them\n",
    "sorted(resdf[(resdf.Altstop>1)].Gene.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Altstop!=1)].sort_values('Gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## CNF02550 hypothetical "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ric8_path = '../GENES/JEC21/CNN01270-t26_1.csv.gz'\n",
    "ric8 = pd.read_csv(ric8_path,index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ric8_name = 'CNN01270'\n",
    "resdf[(resdf.Gene==ric8_name) &(resdf.Altstop>1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "resdf[(resdf.Strain=='CF830') & (resdf.Altstop>1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres = resdf.merge(genes[['Gene','Chromosome','Seqid',\n",
    "                   'Start','End','Strand','Description']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "altstoped = finalres[(finalres.Altstop>1)]\n",
    "altstoped.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "altstoped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres.to_csv('../GENES/gene_changes.csv.gz',index=None)\n",
    "finalres.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres[(finalres.Gene=='CNM00880')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres[(finalres.Gene=='CNJ01150')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres[(finalres.Gene=='CNF04940')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "finalres[(finalres.Gene=='CNJ01150')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test = pd.read_csv('../DATA/OLD/gene_changes.csv.gz')\n",
    "#test.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test[(test.Strain=='Stock6') & (test.Altstop>1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test[(test.Strain=='CF830') & (resdf.Altstop>1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test[(test.Gene==ric8_name)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#a1 = pd.read_csv('../GENES/B3502_A1_Stock1/CNN01270-t26_1.csv.gz',\n",
    "#                 index_col=0)\n",
    "#a1[(a1.Isvar==1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#d1 = pd.read_csv('../GENES/B3502_D1_Stock6/CNN01270-t26_1.csv.gz',\n",
    "#                 index_col=0)\n",
    "#d1[(d1.Isvar==1)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
