{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Bring in needed mods\n",
    "import pandas as pd, numpy as np, glob, sys, os, seaborn as sns\n",
    "from matplotlib import pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "sys.path.insert(0, '/home/croth/QTL-functions/SCRIPTS/')\n",
    "import qtlfunctions as qtlfun"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Set chromosomes\n",
    "chromns = np.arange(1,15)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Nalleles</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Maxlen</th>\n",
       "      <th>Minlen</th>\n",
       "      <th>Type</th>\n",
       "      <th>Callrate</th>\n",
       "      <th>MAF</th>\n",
       "      <th>...</th>\n",
       "      <th>A15</th>\n",
       "      <th>A16</th>\n",
       "      <th>A17</th>\n",
       "      <th>A20</th>\n",
       "      <th>A23</th>\n",
       "      <th>A30</th>\n",
       "      <th>A31</th>\n",
       "      <th>A33</th>\n",
       "      <th>A32</th>\n",
       "      <th>A34</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>21</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1</td>\n",
       "      <td>TTCTC.CTCTG</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>MNP</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.989130</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>317</td>\n",
       "      <td>3.092230e-09</td>\n",
       "      <td>2</td>\n",
       "      <td>CAGT.TAT.TAGT</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>del</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.989130</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>331</td>\n",
       "      <td>3.177140e+02</td>\n",
       "      <td>1</td>\n",
       "      <td>ACAATCCG.ACAAGCCA</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>MNP</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.989130</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>393</td>\n",
       "      <td>3.032550e+03</td>\n",
       "      <td>1</td>\n",
       "      <td>A.G</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>0.98913</td>\n",
       "      <td>0.978261</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>435</td>\n",
       "      <td>2.502520e+03</td>\n",
       "      <td>1</td>\n",
       "      <td>G.C</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>0.98913</td>\n",
       "      <td>0.978261</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 102 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Seqid  Pos          Qual  Nalleles            Alleles  Maxlen  \\\n",
       "1   AE017341.1   21  0.000000e+00         1        TTCTC.CTCTG       5   \n",
       "25  AE017341.1  317  3.092230e-09         2      CAGT.TAT.TAGT       4   \n",
       "26  AE017341.1  331  3.177140e+02         1  ACAATCCG.ACAAGCCA       8   \n",
       "31  AE017341.1  393  3.032550e+03         1                A.G       1   \n",
       "34  AE017341.1  435  2.502520e+03         1                G.C       1   \n",
       "\n",
       "    Minlen Type  Callrate       MAF  ...  A15  A16  A17  A20  A23  A30  A31  \\\n",
       "1        5  MNP   1.00000  0.989130  ...  1.0  1.0  1.0  1.0  1.0  1.0  1.0   \n",
       "25       3  del   1.00000  0.989130  ...  1.0  1.0  1.0  1.0  1.0  1.0  1.0   \n",
       "26       8  MNP   1.00000  0.989130  ...  1.0  1.0  1.0  1.0  1.0  1.0  1.0   \n",
       "31       1  SNP   0.98913  0.978261  ...  1.0  1.0  1.0  1.0  1.0  1.0  1.0   \n",
       "34       1  SNP   0.98913  0.978261  ...  1.0  1.0  1.0  1.0  1.0  1.0  1.0   \n",
       "\n",
       "    A33  A32  A34  \n",
       "1   1.0  1.0  1.0  \n",
       "25  1.0  1.0  1.0  \n",
       "26  1.0  1.0  1.0  \n",
       "31  1.0  1.0  1.0  \n",
       "34  1.0  1.0  1.0  \n",
       "\n",
       "[5 rows x 102 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Gather allelic depth\n",
    "ads = []\n",
    "\n",
    "## Iterate thru chorms\n",
    "for i,c in enumerate(chromns):\n",
    "    \n",
    "    ## Gather paths\n",
    "    info_path = glob.glob('../GENOTYPE/INFO/*.%s.B3502.progeny_info_cols.csv.gz'%c)[0]\n",
    "    dept_path = glob.glob('../GENOTYPE/AF/*.%s.B3502.progeny_allele_ratios.csv.gz'%c)[0]\n",
    "    \n",
    "    ## Load info and depth\n",
    "    info = pd.read_csv(info_path,index_col=0)\n",
    "    dp = pd.read_csv(dept_path,index_col=0)\n",
    "    \n",
    "    ## Concat dataframes\n",
    "    depth = pd.concat([info,dp],axis=1)\n",
    "    \n",
    "    ## Append\n",
    "    ads.append(depth)\n",
    "    \n",
    "## Concat rows\n",
    "ads = pd.concat(ads,axis=0)\n",
    "\n",
    "## View head\n",
    "ads.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Seqid</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Qual</th>\n",
       "      <th>Nalleles</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Maxlen</th>\n",
       "      <th>Minlen</th>\n",
       "      <th>Type</th>\n",
       "      <th>Callrate</th>\n",
       "      <th>MAF</th>\n",
       "      <th>...</th>\n",
       "      <th>A15</th>\n",
       "      <th>A16</th>\n",
       "      <th>A17</th>\n",
       "      <th>A20</th>\n",
       "      <th>A23</th>\n",
       "      <th>A30</th>\n",
       "      <th>A31</th>\n",
       "      <th>A33</th>\n",
       "      <th>A32</th>\n",
       "      <th>A34</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>21</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1</td>\n",
       "      <td>TTCTC.CTCTG</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>MNP</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.989130</td>\n",
       "      <td>...</td>\n",
       "      <td>6.392317</td>\n",
       "      <td>6.266787</td>\n",
       "      <td>5.832890</td>\n",
       "      <td>6.686501</td>\n",
       "      <td>6.894818</td>\n",
       "      <td>4.906891</td>\n",
       "      <td>5.832890</td>\n",
       "      <td>5.700440</td>\n",
       "      <td>5.754888</td>\n",
       "      <td>5.392317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>317</td>\n",
       "      <td>3.092230e-09</td>\n",
       "      <td>2</td>\n",
       "      <td>CAGT.TAT.TAGT</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "      <td>del</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.989130</td>\n",
       "      <td>...</td>\n",
       "      <td>7.357552</td>\n",
       "      <td>7.087463</td>\n",
       "      <td>6.507795</td>\n",
       "      <td>7.357552</td>\n",
       "      <td>7.643856</td>\n",
       "      <td>6.392317</td>\n",
       "      <td>6.228819</td>\n",
       "      <td>6.523562</td>\n",
       "      <td>6.954196</td>\n",
       "      <td>5.832890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>331</td>\n",
       "      <td>3.177140e+02</td>\n",
       "      <td>1</td>\n",
       "      <td>ACAATCCG.ACAAGCCA</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>MNP</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.989130</td>\n",
       "      <td>...</td>\n",
       "      <td>7.238405</td>\n",
       "      <td>6.918863</td>\n",
       "      <td>6.539159</td>\n",
       "      <td>7.303781</td>\n",
       "      <td>7.515700</td>\n",
       "      <td>6.209453</td>\n",
       "      <td>6.169925</td>\n",
       "      <td>6.507795</td>\n",
       "      <td>6.942515</td>\n",
       "      <td>5.832890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>393</td>\n",
       "      <td>3.032550e+03</td>\n",
       "      <td>1</td>\n",
       "      <td>A.G</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>0.98913</td>\n",
       "      <td>0.978261</td>\n",
       "      <td>...</td>\n",
       "      <td>7.285402</td>\n",
       "      <td>6.965784</td>\n",
       "      <td>6.285402</td>\n",
       "      <td>7.257388</td>\n",
       "      <td>7.546894</td>\n",
       "      <td>6.409391</td>\n",
       "      <td>6.321928</td>\n",
       "      <td>6.392317</td>\n",
       "      <td>6.820179</td>\n",
       "      <td>6.129283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>AE017341.1</td>\n",
       "      <td>435</td>\n",
       "      <td>2.502520e+03</td>\n",
       "      <td>1</td>\n",
       "      <td>G.C</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>0.98913</td>\n",
       "      <td>0.978261</td>\n",
       "      <td>...</td>\n",
       "      <td>7.294621</td>\n",
       "      <td>7.076816</td>\n",
       "      <td>6.643856</td>\n",
       "      <td>7.209453</td>\n",
       "      <td>7.562242</td>\n",
       "      <td>6.339850</td>\n",
       "      <td>6.442943</td>\n",
       "      <td>6.459432</td>\n",
       "      <td>6.832890</td>\n",
       "      <td>6.189825</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 102 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Seqid  Pos          Qual  Nalleles            Alleles  Maxlen  \\\n",
       "1   AE017341.1   21  0.000000e+00         1        TTCTC.CTCTG       5   \n",
       "25  AE017341.1  317  3.092230e-09         2      CAGT.TAT.TAGT       4   \n",
       "26  AE017341.1  331  3.177140e+02         1  ACAATCCG.ACAAGCCA       8   \n",
       "31  AE017341.1  393  3.032550e+03         1                A.G       1   \n",
       "34  AE017341.1  435  2.502520e+03         1                G.C       1   \n",
       "\n",
       "    Minlen Type  Callrate       MAF  ...       A15       A16       A17  \\\n",
       "1        5  MNP   1.00000  0.989130  ...  6.392317  6.266787  5.832890   \n",
       "25       3  del   1.00000  0.989130  ...  7.357552  7.087463  6.507795   \n",
       "26       8  MNP   1.00000  0.989130  ...  7.238405  6.918863  6.539159   \n",
       "31       1  SNP   0.98913  0.978261  ...  7.285402  6.965784  6.285402   \n",
       "34       1  SNP   0.98913  0.978261  ...  7.294621  7.076816  6.643856   \n",
       "\n",
       "         A20       A23       A30       A31       A33       A32       A34  \n",
       "1   6.686501  6.894818  4.906891  5.832890  5.700440  5.754888  5.392317  \n",
       "25  7.357552  7.643856  6.392317  6.228819  6.523562  6.954196  5.832890  \n",
       "26  7.303781  7.515700  6.209453  6.169925  6.507795  6.942515  5.832890  \n",
       "31  7.257388  7.546894  6.409391  6.321928  6.392317  6.820179  6.129283  \n",
       "34  7.209453  7.562242  6.339850  6.442943  6.459432  6.832890  6.189825  \n",
       "\n",
       "[5 rows x 102 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Gather depths\n",
    "dps = []\n",
    "\n",
    "## Loop thru chroms\n",
    "for i,c in enumerate(chromns):\n",
    "    \n",
    "    ## Get paths\n",
    "    info_path = glob.glob('../GENOTYPE/INFO/*.%s.B3502.progeny_info_cols.csv.gz'%c)[0]\n",
    "    dept_path = glob.glob('../GENOTYPE/DP/*.%s.B3502.progeny_depths.csv.gz'%c)[0]\n",
    "    \n",
    "    ## Load df\n",
    "    info = pd.read_csv(info_path,index_col=0)\n",
    "    dp = np.log2(pd.read_csv(dept_path,index_col=0)+1)\n",
    "    \n",
    "    ## Concat columns\n",
    "    depth = pd.concat([info,dp],axis=1)\n",
    "    \n",
    "    ## Append dataframe\n",
    "    dps.append(depth)\n",
    "    \n",
    "## Concat rows\n",
    "dps = pd.concat(dps,axis=0)\n",
    "\n",
    "## View head\n",
    "dps.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92\n"
     ]
    }
   ],
   "source": [
    "## Gather sample names and print length\n",
    "samples = np.unique(dp.columns)\n",
    "print(len(samples))\n",
    "\n",
    "## Gather progeny\n",
    "progeny = sorted([s for s in samples if len(s)<=4])\n",
    "\n",
    "## Gather ancestors\n",
    "ancestors = sorted([s for s in samples if len(s)>4])\n",
    "\n",
    "## Check work\n",
    "assert len(progeny) + len(ancestors) == len(samples)\n",
    "\n",
    "## Gather B3502 stocks\n",
    "b3502 = [s for s in ancestors if s.split(\"_\")[0]=='B3502']\n",
    "\n",
    "## parents and progenitor strains\n",
    "progenitors = [s for s in ancestors if s not in b3502]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "## For each sample, plot the depth and allelic depth ratio\n",
    "for sample in samples:\n",
    "\n",
    "    ## Call figure\n",
    "    fig,ax = plt.subplots(2,1,figsize=(12,5),sharex=True)\n",
    "    fig.set_facecolor('w')\n",
    "\n",
    "    ## Set axis and title\n",
    "    plt.sca(ax[0])\n",
    "    plt.title(sample)\n",
    "\n",
    "    ## plot log2 depth\n",
    "    qtlfun.manhattan(dps,pval=sample,chrom='Seqid',xlabel='',\n",
    "                 ylabel='log$_2$ (Depth)',ylims=(0,12),ax=ax[0]);\n",
    "\n",
    "    ## plot allelic depth ratio\n",
    "    qtlfun.manhattan(ads,pval=sample,chrom='Seqid',chromlabels=chromns,\n",
    "                 ylabel='Allelic Depth',ylims=(0,1),ax=ax[1]);\n",
    "    \n",
    "    ## Save figure and close figure\n",
    "    plt.savefig('../FIGURES/DEPTHPLOTS/%s.png'%sample,\n",
    "                dpi=100,bbox_inches='tight')\n",
    "    plt.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
